DevShell/skaffold-nix-builder.nix

```diff
@@ -0,0 +1,16 @@
+{ pkgs }:
+with pkgs;
+
+writeShellScriptBin "skaffold-nix-builder" ''
+  set -e
+
+  repo=$(echo "$IMAGE" | cut -d':' -f 1)
+  tag=$(echo "$IMAGE" | cut -d':' -f 2)
+  out="$(mktemp -u)"
+  nix-build "$BUILD_CONTEXT/Nixfile.nix" -A "images.$repo" -o "$out" --arg image-tag "\"$tag\""
+  ${docker}/bin/docker load -i $out
+
+  if $PUSH_IMAGE; then
+      ${docker}/bin/docker push $IMAGE
+  fi
+''
```

Comment:
nix is the best tool to build reproducible containers images in a minimal and cach-optimizaed fashion and does not have several of the shortcomings that `Dockerfiles` have (reproducability issues with `apt-get`, build time dependencies in final images if not carefully crafted, bloated with a lot of unneeded baggage if using one of the more popular base images, cache inefficiencies &mdash; nix images have a "popularity algorithm" that tends to put "popular" dependencies, like libgc, within their own caching layer &rarr; better startup times). So I built a little `skaffold` custom builder to build minimal and cache efficient nix images.